cli/stream_command.go

```diff
@@ -1403,18 +1419,53 @@ func (c *streamCmd) copyAndEditStream(cfg api.StreamConfig, pc *fisk.ParseContex
 		return cfg, fmt.Errorf("invalid compression algorithm")
 	}
 
-	if c.repubDest != "" && c.repubSource != "" {
-		cfg.RePublish = &api.RePublish{
-			Source:      c.repubSource,
-			Destination: c.repubDest,
-			HeadersOnly: c.repubHeadersOnly,
-		}
+	var repubConfig api.RePublish
+
+	if cfg.RePublish != nil {
+		repubConfig = *cfg.RePublish
+	}
+
+	if c.repubSource != "" {
+		repubConfig.Source = c.repubSource
+	}
+
+	if c.repubDest != "" {
+		repubConfig.Destination = c.repubDest
+	}
+
+	if c.repubHeadersOnly {
+		repubConfig.HeadersOnly = true
+	} else if repubConfig.HeadersOnly {
+		repubConfig.HeadersOnly = false
+	}
+
+	if (repubConfig.Source == "" || repubConfig.Source == ">") && repubConfig.Destination == "" {
+		cfg.RePublish = nil
```

Comment:
still weird to me to make all the repubConfig stuff and then not use it later on. Ditto later with SubjectTransform

Both would be easier with a func helper so you can just return early from the func in this case

Reply:
Yes, that was convoluted and had some unneeded checks, and --no-republish/--no-transform had been tacked on at the end, rewrote for both it's much cleaner now thanks for that.